src/main/java/application/repository/BookRepository.java

```diff
@@ -0,0 +1,13 @@
+package application.repository;
+
+import application.model.Book;
+import java.util.List;
+import java.util.Optional;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    List<Book> findAll();
+
+    Optional<Book> findBookById(Long id);

```

Comment:
```suggestion
    Optional<Book> findById(Long id);
```
Also, it is more than the task requirements.